src/webhook/mutation/pod/dataingest/mutator.go

```diff
@@ -0,0 +1,103 @@
+package dataingest_mutation
+
+import (
+	dtingestendpoint "github.com/Dynatrace/dynatrace-operator/src/ingestendpoint"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/src/webhook"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DataIngestPodMutator struct {
+	webhookNamespace string
+	client           client.Client
+	metaClient       client.Client
```

Comment:
why is this needed?

Reply:
finding the workload info (if I remember correctly its faster, as we only care about the metadata)